maven/bnd-export-maven-plugin/src/main/java/aQute/bnd/maven/export/plugin/ExportMojo.java

```diff
@@ -65,7 +66,7 @@ private void resolve(StandaloneRun run) throws Exception, IOException {
 			List<Container> runBundles = projectResolver.getRunBundles();
 			Collection<Container> currentRunBundles = run.getRunbundles();
 			if (!CollectionUtils.isEqualCollection(currentRunBundles, runBundles)) {
-				if (failOnChanges) {
+				if (!resolve && failOnChanges) {
```

Comment:
I guess I missunderstood what failOnChanges was meant to do. I thought it was meant to be used if you do not want to resolve.

If it only makes sense if you resolve then if (failOnChanges) is correct. In this case the user has to set resolve to true and failOnChanges to false to do a fully automatic resolve and export.